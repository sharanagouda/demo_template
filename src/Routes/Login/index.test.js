import { Login } from "./index";

jest.mock("@discovery-dni/cd-tv.platform", () => ({
  androidKeyboardVisible$: {
    subscribe: jest.fn()
  }
}));

describe("user is logged in", () => {
  let onLogoutClickMock;

  beforeAll(() => {
    onLogoutClickMock = jest.fn();
  });

  test("snapshot matches", () => {
    const component = new Login();
    component.data = { loggedIn: true };
    component.render();

    expect(component).toMatchSnapshot();
  });

  test("signout click", () => {
    const component = new Login();
    component.data = { loggedIn: true };
    component.onLogoutClick = onLogoutClickMock;
    component.render();

    const logoutBtn = component.logoutButton.element;
    logoutBtn.click();

    expect(onLogoutClickMock).toHaveBeenCalled();
  });
});

describe("user is not logged in", () => {
  let onSigninButtonClickMock;

  beforeAll(() => {
    onSigninButtonClickMock = jest.fn();
    global.Headers = () => {};
  });

  test("snapshot matches", () => {
    const component = new Login();
    component.data = { loggedIn: false };
    component.render();

    expect(component).toMatchSnapshot();
  });

  test("login click", () => {
    const component = new Login();
    component.data = { loggedIn: false };
    component.onSigninButtonClick = onSigninButtonClickMock;
    component.render();

    const loginBtn = component.loginButton.element;
    loginBtn.click();

    expect(onSigninButtonClickMock).toHaveBeenCalled();
  });
});
