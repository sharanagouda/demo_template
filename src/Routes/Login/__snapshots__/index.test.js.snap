// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user is logged in snapshot matches 1`] = `
Login {
  "continueBrowsing": Object {
    "addClass": [Function],
    "appendTo": [Function],
    "element": <a
      class="button no-login"
      href="undefined"
      replace="true"
      tabindex="0"
    >
      Browse content
    </a>,
    "isMounted": false,
    "prependTo": [Function],
    "remove": [Function],
    "removeClass": [Function],
  },
  "data": Object {
    "loggedIn": true,
  },
  "logoutButton": Object {
    "addClass": [Function],
    "appendTo": [Function],
    "element": <div
      class="button"
      tabindex="0"
    >
      Sign out
    </div>,
    "isMounted": false,
    "prependTo": [Function],
    "remove": [Function],
    "removeClass": [Function],
  },
  "onAndroidFocusChange": [Function],
  "onSigninButtonClick": [Function],
  Symbol(cd-tv-router-subscriptions): Array [],
}
`;

exports[`user is not logged in snapshot matches 1`] = `
Login {
  "continueBrowsing": Object {
    "addClass": [Function],
    "appendTo": [Function],
    "element": <a
      class="button no-login"
      href="undefined"
      replace="true"
      tabindex="0"
    >
      Browse content
    </a>,
    "isMounted": false,
    "prependTo": [Function],
    "remove": [Function],
    "removeClass": [Function],
  },
  "data": Object {
    "loggedIn": false,
  },
  "emailContainer": Object {
    "addClass": [Function],
    "appendTo": [Function],
    "element": <div
      class="login-input"
    >
      <span
        class="login-label"
      >
        Email:
      </span>
      <input
        class="input"
        tabindex="0"
      />
    </div>,
    "isMounted": false,
    "prependTo": [Function],
    "remove": [Function],
    "removeClass": [Function],
  },
  "emailInput": Object {
    "addClass": [Function],
    "appendTo": [Function],
    "element": <input
      class="input"
      tabindex="0"
    />,
    "isMounted": false,
    "prependTo": [Function],
    "remove": [Function],
    "removeClass": [Function],
  },
  "emailInputError": Object {
    "addClass": [Function],
    "appendTo": [Function],
    "element": <span
      class="error"
    >
      Something went wrong
    </span>,
    "isMounted": false,
    "prependTo": [Function],
    "remove": [Function],
    "removeClass": [Function],
  },
  "loginButton": Object {
    "addClass": [Function],
    "appendTo": [Function],
    "element": <div
      class="button"
      tabindex="0"
    >
      Login
    </div>,
    "isMounted": false,
    "prependTo": [Function],
    "remove": [Function],
    "removeClass": [Function],
  },
  "onAndroidFocusChange": [Function],
  "onSigninButtonClick": [Function],
  "passwordContainer": Object {
    "addClass": [Function],
    "appendTo": [Function],
    "element": <div
      class="login-input"
    >
      <span
        class="login-label"
      >
        Password
      </span>
      <input
        class="input"
        tabindex="0"
      />
    </div>,
    "isMounted": false,
    "prependTo": [Function],
    "remove": [Function],
    "removeClass": [Function],
  },
  "passwordInput": Object {
    "addClass": [Function],
    "appendTo": [Function],
    "element": <input
      class="input"
      tabindex="0"
    />,
    "isMounted": false,
    "prependTo": [Function],
    "remove": [Function],
    "removeClass": [Function],
  },
  "passwordInputError": Object {
    "addClass": [Function],
    "appendTo": [Function],
    "element": <span
      class="error"
    >
      To reset your password, please go to the website.
    </span>,
    "isMounted": false,
    "prependTo": [Function],
    "remove": [Function],
    "removeClass": [Function],
  },
  Symbol(cd-tv-router-subscriptions): Array [],
}
`;
